homeassistant/components/spotify/__init__.py

```diff
@@ -52,6 +61,20 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     return True
 
 
+async def async_migrate_entry(hass, entry):
+    """Handle migration of a previous version config entry."""
+    if entry.version == 1:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": "reauth"},
+                data=entry.data,
+            )
+        )
+
+    return True
```

Comment:
Can we do it without a migration and just checking that the required tokens are in the token info? Because by returning True the config entry will still continue to set up, so there is no difference except that with a version 2, you will not be able to roll back to an older Home Assistant without reconfiguring Spotify (even if our token has more scopes) because HA will reject a config entry version it doesn't know.

Reply:
Honestly, that is a design flaw in the migration. We've been doing a lot of migrations outside of the migration handling, which because of the above-listed reason. Instead of having handling like this centralized, we now are polluting all entry setups until eternity with "migrations".

Reply:
The old version of the Spotify integration will not know that it is actually compatible with v2. Since the config entry is a different version, it will look for the migrate function to make it compatible, but that function is not there in the old HA.

Reply:
Yeah just tested it, it just bails out indeed, because it expected a migration that isn't there.

Reply:
Done.